qiskit/optimization/converters/quadratic_program_to_operator.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+
+"""The converter from an ```QuadraticProgram``` to ``Operator``."""
+
+from typing import Tuple
+
+import numpy as np
+from qiskit.quantum_info import Pauli
+
+from qiskit.aqua.operators import WeightedPauliOperator
+
+from ..problems.quadratic_program import QuadraticProgram
+from ..exceptions.qiskit_optimization_error import QiskitOptimizationError
+
+
+class QuadraticProgramToOperator:
+    """Convert an optimization problem into a qubit operator."""
+
+    def __init__(self) -> None:
+        """Initialize the internal data structure."""
+        self._src = None
+
+    def encode(self, op: QuadraticProgram) -> Tuple[WeightedPauliOperator, float]:
```

Comment:
I think at this level the use of the existing WeightedPauliOperator versus the new logic is not so exposed to user such that when this becomes the new operator hopefully it won't have too much external impact. Z2 Symmetry function, that is not yet in new operator logic, prevents chemistry moving over now. I do not think that lack would prevent this right - I have not seen it used with the Ising H's but I guess it could be.